src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/HostDiagnosticAnalyzerProvider.cs

```diff
@@ -17,7 +18,16 @@ internal sealed class HostDiagnosticAnalyzerProvider(string? razorSourceGenerato
     {
         if (File.Exists(razorSourceGenerator))
         {
-            return [(razorSourceGenerator, ProjectSystemProject.RazorVsixExtensionId)];
+            // we also have to redirect the utilities and object pool assemblies
+            var razorDir = Path.GetDirectoryName(razorSourceGenerator) ?? "";
```

Comment:
What is the CWD for the OOP process? Cause I believe the code below will end up using CWD to resolve the relative paths. Is that going to work or fail due to missing files, etc ... Cause if it's going to fail would prefer we just throw an exception with a good message here.

Reply:
Made it an exception if we can't find the dependencies